pkg/sql/importer/import_job.go

```diff
@@ -1220,14 +1220,20 @@ func ingestWithRetry(
 	var err error
 	var retryCount int32
 	for r := retry.StartWithCtx(ctx, retryOpts); r.Next(); {
-		retryCount++
-		resumerSpan.RecordStructured(&roachpb.RetryTracingEvent{
-			Operation:     "importResumer.ingestWithRetry",
-			AttemptNumber: retryCount,
-			RetryError:    tracing.RedactAndTruncateError(err),
-		})
-		res, err = distImport(ctx, execCtx, job, tables, typeDescs, from, format, walltime,
-			alwaysFlushProgress, procsPerNode)
+		for {
+			retryCount++
+			resumerSpan.RecordStructured(&roachpb.RetryTracingEvent{
+				Operation:     "importResumer.ingestWithRetry",
+				AttemptNumber: retryCount,
+				RetryError:    tracing.RedactAndTruncateError(err),
+			})
+			res, err = distImport(ctx, execCtx, job, tables, typeDescs, from, format, walltime,
+				alwaysFlushProgress, procsPerNode)
+			// Replanning errors should not count towards retry limits.
```

Comment:
This seems reasonable, but do we need something to protect against 'replanning thrashing'.  Perhaps we could pass in the "last replan" time and disallow replanning too often within some time period?

Reply:
the replan trigger owns that -- e.g. in the import change by default it won't do more than once every 2min.